tests/unit/test_solvers/test_processed_variable.py

```diff
@@ -28,9 +28,11 @@ def to_casadi(var_pybamm, y, inputs=None):
 
 
 def process_and_check_2D_variable(
-    var, first_spatial_var, second_spatial_var, disc=None, geometry_options={}
+    var, first_spatial_var, second_spatial_var, disc=None, geometry_options=None
 ):
     # first_spatial_var should be on the "smaller" domain, i.e "r" for an "r-x" variable
+    if geometry_options is None:
+        geometry_options = {}
```

Comment:
The violated rule is [mutable-argument-default (B006)](https://docs.astral.sh/ruff/rules/mutable-argument-default/), basically function defaults are evaluated once so if you have a mutable object, every time you modify it, its modifications will persist across calls. It's to avoid unexpected behavior, although we are in unit tests.

If we want we could ignore it using `# noqa: B006`:

```python
def process_and_check_2D_variable(
    var, first_spatial_var, second_spatial_var, disc=None, geometry_options={}  # noqa: B006
):
```

Or if we want to ignore it at project level, we could add it in `pyproject.toml`:
```toml
ignore = [
    # ...
    "B006",        # Uses of mutable objects as function argument defaults
]
```